test/types/index.ts

```diff
@@ -138,27 +138,41 @@ server.addHook('onClose', (instance, done) => {
 server.addHook('onRoute', (opts) => {
 })
 
-const opts: fastify.RouteShorthandOptions<http2.Http2Server, http2.Http2ServerRequest, http2.Http2ServerResponse> = {
-  schema: {
-    response: {
-      200: {
-        type: 'object',
-        properties: {
-          hello: {
-            type: 'string'
-          }
+const schema: fastify.RouteSchema = {
+  body: {
+    type: 'object'
+  },
+  querystring: {
+    type: 'object'
+  },
+  params: {
+    type: 'object'
+  },
+  headers: {
+    type: 'object'
+  },
+  response: {
+    200: {
+      type: 'object',
+      properties: {
+        hello: {
+          type: 'string'
         }
-      },
-      '2xx': {
-        type: 'object',
-        properties: {
-          hello: {
-            type: 'string'
-          }
+      }
+    },
+    '2xx': {
+      type: 'object',
+      properties: {
+        hello: {
+          type: 'string'
         }
       }
     }
-  },
+  }
+}
+
+const opts: fastify.RouteShorthandOptions<http2.Http2Server, http2.Http2ServerRequest, http2.Http2ServerResponse> = {
+  schema,
```

Comment:
Per ["node": ">=6"](https://github.com/fastify/fastify/blob/master/package.json#L75) ran tests using Node v6.16.0 also. So using shorthand property seems okay in this case. Please let me know if changes are required.